src/main/kotlin/com/chattriggers/ctjs/api/world/Scoreboard.kt

```diff
@@ -99,27 +103,40 @@ object Scoreboard {
      * @param override whether to remove old lines with the same score
      */
     @JvmStatic
-    fun setLine(score: Int, line: String, override: Boolean) {
+    @JvmOverloads
+    fun setLine(score: Int, line: TextComponent, override: Boolean = false) {
         val scoreboard = toMC() ?: return
         val sidebarObjective = getSidebar() ?: return
 
-        val scores = scoreboard.getAllPlayerScores(sidebarObjective)
-
         if (override) {
-            scores.filter {
-                it.score == score
+            //#if MC>=12004
+            scoreboard.getScoreboardEntries(sidebarObjective).filter {
+                it.value == score
             }.forEach {
-                scoreboard.resetPlayerScore(it.playerName, sidebarObjective)
+                scoreboard.removeScore({ it.owner }, sidebarObjective)
             }
+            //#else
+            //$$ scoreboard.getAllPlayerScores(sidebarObjective).filter {
+            //$$     it.score == score
+            //$$ }.forEach {
+            //$$     scoreboard.resetPlayerScore(it.playerName, sidebarObjective)
+            //$$ }
+            //#endif
         }
 
-        scoreboard.getPlayerScore(line, sidebarObjective).score = score
+        //#if MC>=12004
+        scoreboard.knownScoreHolders.forEach {
+            val scoreboardScore = scoreboard.getScore({ it.nameForScoreboard }, sidebarObjective) as? ScoreboardScore
+            if (scoreboardScore?.score == score)
+                scoreboardScore.displayText = line
+        }
+        //#else
+        //$$ scoreboard.getPlayerScore(line.formattedText, sidebarObjective).score = score
```

Comment:
This definitely doesn't work, but I'm not quite sure how to fix it, since the player name is immutable in 1.19